sdk/trace/time.go

```diff
@@ -0,0 +1,54 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package trace
+
+import (
+	"time"
+)
+
+// Clock is the entrypoint for providing time to span's start/end timestamp.
+// By default the standard "time" package will be used. User can replace
+// it with customized clock implementation (e.g. has additional clock
+// synchronization logic) by using the `WithClock` option.
+type Clock interface {
```

Comment:
Do we need a Clock interface? It seems like just a Stopwatch interface would be sufficient to implement new clocks reporting a start time and the elapsed time.

Reply:
We need some way to generate the `Stopwatch`. There will be one stopwatch per span.

Reply:
What's the advantage of this solution compared to a single interface returning the actual time and leaving the calculation of the elapsed time to the caller?

Reply:
To better utilize this custom clock sometimes a "state" needs to be kept during the span. Say an "offset" clock which will add an offset to both the start time and end time (and the offset value is updated somewhere else periodically, e.g. via NTP). In this case it would be better to make sure the same "offset" is applied to both the start time and end time in a span even the offset is changed during the span.

Reply:
I see. Thanks for the explanation.